python/ray/ml/result.py

```diff
@@ -4,15 +4,33 @@
 
 
 class Result:
-    """The final result of a ML training run.
+    """The final result of a ML training run or a Tune trial.
 
-    This is the class produced by Trainer.fit(). It contains a checkpoint, which
-    can be used for resuming training and for creating a Predictor object. It also
-    contains a metrics object describing training metrics.
+    This is the class produced by Trainer.fit() or Tuner.fit() (through ResultGrid).
+    It contains a checkpoint, which can be used for resuming training and for
+    creating a Predictor object. It also contains a metrics object describing
+    training metrics. `status` and `error` are included so that non successful runs
+    and trials can be represented as well.
+
+    Args:
+        metrics: The final metrics as reported by an Trainable.
+        checkpoint: The final checkpoint of the Trainable.
+        status: The status of the Trainable run. Should be between TERMINATED and ERROR.
+        error: The execution error of the Trainable run, when `status` is reported as
```

Comment:
Nit: Add type hints to arg descriptions

Reply:
See discussion here: https://anyscaleteam.slack.com/archives/CLJQR6420/p1646858352916369